### PR TITLE
Hover stylings

### PR DESCRIPTION
**Pull request by mclin**
Made it easier to pass in stylings, and added hover state param to 'value', 'nestedNode' and 'rootNode' stylings.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
     "rimraf": "^2.5.2"
   },
   "peerDependencies": {
-    "react": "^0.14.0 || ^15.0.0-0"
+    "react": "^0.14.0 || ^15.0.0-0",
+    "react-dom": "^0.14.0 || ^15.0.0-0"
```
Comment (alexkuz) [inline package.json]: Not sure why we need it, but fine :)

File: src/JSONNestedNode.js [hunk 2]
```diff
@@ -92,9 +92,14 @@ export default class JSONNestedNode extends React.Component {
     // calculate individual node expansion if necessary
     const expanded = props.shouldExpandNode && !props.isCircular ?
         props.shouldExpandNode(props.keyPath, props.data, props.level) : false;
+
+    this.handleMouseOver = this.handleMouseOver.bind(this);
+    this.handleMouseOut = this.handleMouseOut.bind(this);
```
Comment (alexkuz) [inline src/JSONNestedNode.js]: Can you please rewrite this as class properties?

File: src/JSONValueNode.js [hunk 3]
```diff
@@ -4,41 +4,57 @@ import React, { PropTypes } from 'react';
...
+  };
 
-JSONValueNode.defaultProps = {
-  valueGetter: value => value
-};
+  constructor() {
+    super();
+    this.state = { hover: false };
+    this.handleMouseOver = this.handleMouseOver.bind(this);
+    this.handleMouseOut = this.handleMouseOut.bind(this);
```
Comment (alexkuz) [inline src/JSONValueNode.js]: Same here (so there's no need for constructor, just define `state` as property, as well)

File: src/index.js [hunk 4]
```diff
@@ -88,7 +88,8 @@ export default class JSONTree extends React.Component {
       ...rest
     } = this.props;
 
-    const styling = createStylingFromTheme(checkLegacyTheming(theme, rest), invertTheme);
+    const styling = rest.styling ||
```
Comment (alexkuz) [inline src/index.js]: There's no need for this, actually, as styling can be overridden in `theme`.

**Comment by alexkuz** [comment]
@mclin thanks for PR! There are some minor code style issues, otherwise looks fine for me.

File: package.json (hunk 1 above)
Comment (mclin) [inline package.json]: Removed it. Thought I needed it for some reason, but it's only in the example

File: src/JSONNestedNode.js (hunk 2 above)
Comment (mclin) [inline src/JSONNestedNode.js]: Done.

File: src/JSONValueNode.js (hunk 3 above)
Comment (mclin) [inline src/JSONValueNode.js]: done

File: src/index.js (hunk 4 above)
Comment (mclin) [inline src/index.js]: Got it. After messing around I get how this works now. Removed this change.

**Pull request closed without merging**